[WK2] Stop relying on WebKitLegacy code to do geolocation permission prompting

cdumez opened the pull request:
#### bbb71e0a35315e1d6b3df324925377bf7b55a789
<pre>
[WK2] Stop relying on WebKitLegacy code to do geolocation permission prompting
<a href="https://bugs.webkit.org/show_bug.cgi?id=251368">https://bugs.webkit.org/show_bug.cgi?id=251368</a>
rdar://102728659

Reviewed by Geoffrey Garen.

Stop relying on WebKitLegacy code to do geolocation permission prompting in
WebKit2. We now have a WKWebGeolocationPolicyDecider class that lives in
the WebKit2 project.

Also enhance the prompting dialog to be closer to the one MobileSafari uses
(see screenshots on radar).

The logic for remembering the geolocation permission is identical to the
one WebKitLegacy is using, except that we only use the host as key instead
of the full security origin. Note that the prompt only shows the host, like
it does in MobileSafari.

* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/ios/WKGeolocationProviderIOS.mm:
(-[WKGeolocationProviderIOS geolocationAuthorizationGranted]):
(-[WKWebAllowDenyPolicyListener deny]):
(-[WKWebAllowDenyPolicyListener denyOnlyThisRequest]): Deleted.
(-[WKWebAllowDenyPolicyListener shouldClearCache]): Deleted.
* Source/WebKit/UIProcess/ios/WKGeolocationProviderIOSObjCSecurityOrigin.mm: Removed.
* Source/WebKit/UIProcess/ios/WKWebGeolocationPolicyDecider.h: Added.
* Source/WebKit/UIProcess/ios/WKWebGeolocationPolicyDeciderIOS.mm: Added.
(clearGeolocationCache):
(appHasPreciseLocationPermission):
(appDisplayName):
(getToken):
(PermissionRequest::create):
(+[WKWebGeolocationPolicyDecider sharedPolicyDecider]):
(-[WKWebGeolocationPolicyDecider init]):
(-[WKWebGeolocationPolicyDecider dealloc]):
(-[WKWebGeolocationPolicyDecider decidePolicyForGeolocationRequestFromOrigin:requestingURL:view:listener:]):
(-[WKWebGeolocationPolicyDecider _executeNextChallenge]):
(-[WKWebGeolocationPolicyDecider _finishActiveChallenge:]):
(-[WKWebGeolocationPolicyDecider clearCache]):
(-[WKWebGeolocationPolicyDecider _siteFileInContainerDirectory:creatingIntermediateDirectoriesIfNecessary:]):
(-[WKWebGeolocationPolicyDecider _siteFile]):
(createChallengeDictionary):
(-[WKWebGeolocationPolicyDecider _loadWithCompletionHandler:]):
(-[WKWebGeolocationPolicyDecider _save]):
(-[WKWebGeolocationPolicyDecider _getChallengeCountFromHistoryForToken:requestingURL:]):
(-[WKWebGeolocationPolicyDecider _addChallengeCount:forToken:requestingURL:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/259658@main">https://commits.webkit.org/259658@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/152d1cdd4741e416ac49391b1a43c9c887005a9e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/105496 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/14601 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/38409 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/114756 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/174908 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/109396 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16015 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/5829 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/97802 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/114604 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111253 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/12181 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95169 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/39669 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/94045 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/26805 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/81360 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/7881 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28161 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/8234 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/4750 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14009 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/47704 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/9908 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3570 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren reviewed (approved):
r=me

If an app uses both WK2 and WK1, will they end up sharing the same on-disk record? (I think it would be good if they did -- not super important, but it makes it easier to clear website data appropriately.)

cdumez committed:
[WK2] Stop relying on WebKitLegacy code to do geolocation permission prompting
https://bugs.webkit.org/show_bug.cgi?id=251368
rdar://102728659

Reviewed by Geoffrey Garen.

Stop relying on WebKitLegacy code to do geolocation permission prompting in
WebKit2. We now have a WKWebGeolocationPolicyDecider class that lives in
the WebKit2 project.

Also enhance the prompting dialog to be closer to the one MobileSafari uses
(see screenshots on radar).

The logic for remembering the geolocation permission is identical to the
one WebKitLegacy is using, except that we only use the host as key instead
of the full security origin. Note that the prompt only shows the host, like
it does in MobileSafari.

* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/ios/WKGeolocationProviderIOS.mm:
(-[WKGeolocationProviderIOS geolocationAuthorizationGranted]):
(-[WKWebAllowDenyPolicyListener deny]):
(-[WKWebAllowDenyPolicyListener denyOnlyThisRequest]): Deleted.
(-[WKWebAllowDenyPolicyListener shouldClearCache]): Deleted.
* Source/WebKit/UIProcess/ios/WKGeolocationProviderIOSObjCSecurityOrigin.mm: Removed.
* Source/WebKit/UIProcess/ios/WKWebGeolocationPolicyDecider.h: Added.
* Source/WebKit/UIProcess/ios/WKWebGeolocationPolicyDeciderIOS.mm: Added.
(clearGeolocationCache):
(appHasPreciseLocationPermission):
(appDisplayName):
(getToken):
(PermissionRequest::create):
(+[WKWebGeolocationPolicyDecider sharedPolicyDecider]):
(-[WKWebGeolocationPolicyDecider init]):
(-[WKWebGeolocationPolicyDecider dealloc]):
(-[WKWebGeolocationPolicyDecider decidePolicyForGeolocationRequestFromOrigin:requestingURL:view:listener:]):
(-[WKWebGeolocationPolicyDecider _executeNextChallenge]):
(-[WKWebGeolocationPolicyDecider _finishActiveChallenge:]):
(-[WKWebGeolocationPolicyDecider clearCache]):
(-[WKWebGeolocationPolicyDecider _siteFileInContainerDirectory:creatingIntermediateDirectoriesIfNecessary:]):
(-[WKWebGeolocationPolicyDecider _siteFile]):
(createChallengeDictionary):
(-[WKWebGeolocationPolicyDecider _loadWithCompletionHandler:]):
(-[WKWebGeolocationPolicyDecider _save]):
(-[WKWebGeolocationPolicyDecider _getChallengeCountFromHistoryForToken:requestingURL:]):
(-[WKWebGeolocationPolicyDecider _addChallengeCount:forToken:requestingURL:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/259658@main
